### ISSUE TITLE
User Action not being merged after sign in

### ISSUE THREAD

**Issue by twistershark** [Stand-With-Crypto/swc-web#1278]
Currently, if you clear all cache and app data to simulate a new user accessing SWC, do a random action like 'Follow us on X' and clicks on Sign In on the modal that will show up, there is a problem where the action that you just completed is not merged to that user after sign in. 

The weird UX is that the user is seeing, after the steps above, the success screen saying the action was completed, but is isn't marked as complete on the list of user action. Prob the action was created but it was not assigned to the user after login.

**Comment by twistershark** [comment]
**TL;DR; There is no fix to this because it is not a problem, but could we change the logic to be able to merge only the user action that was completed?**

After investigation, I believe that there is nothing wrong with the current flow. Here is why:

The scenario described in this issue will only happen if the users completes an action and then tries to log in with an existing user. This won't trigger the accounts merge logic and by not doing it, the user will experience the scenario described.

But, if the user tries to login, after completing the action, with a new user, the merge logic will happen and the completed action will be associated with this user record created.

---

I know that we don't merge user data in this situation because of security reasons, but what if we associate this specific user action with the authenticated user and remove it from the unauthenticated user if the user logged in immediately on the post action screen?

If would look something like this: Completes an action while unauthenticated -> sees the post action log in section -> log in -> we change the user action's associated user with the new authenticated user if the authenticated user already exists.



@travisbloom-cb
